modules/wkb/src/lib/parse-wkb.js

```diff
@@ -0,0 +1,220 @@
+/* eslint-disable */
+
+
+const NUM_DIMENSIONS = {
+  0: 2, // 2D
+  1: 3, // 3D (Z)
+  2: 3, // 3D (M)
+  3: 4 // 4D (ZM)
+};
+
+export default function parseWKB(buffer) {
+  const view = new DataView(buffer);
+  let offset = 0;
+
+  // Check endianness of data
+  const littleEndian = view.getUint8(offset) === 1;
+  offset++;
+
+  // 4-digit code representing dimension and type of geometry
+  const geometryCode = view.getUint32(offset, littleEndian);
+  offset += 4;
+
+  const geometryType = geometryCode % 1000;
+  const dimension = NUM_DIMENSIONS[(geometryCode - geometryType) / 1000];
+
+  switch (geometryType) {
+    case 1:
+      return parsePoint(view, offset, dimension, littleEndian);
+      break;
+    case 2:
+      return parseLineString(view, offset, dimension, littleEndian);
+      break;
+    case 3:
+      return parsePolygon(view, offset, dimension, littleEndian);
+      break;
+    case 4:
+      return parseMultiPoint(view, offset, dimension, littleEndian);
+      break;
+    case 5:
+      return parseMultiLineString(view, offset, dimension, littleEndian);
+      break;
+    case 6:
+      return parseMultiPolygon(view, offset, dimension, littleEndian);
+      break;
+    case 7:
+      // TODO: handle GeometryCollections
+      // return parseGeometryCollection(view, offset, dimension, littleEndian);
+      break;
+    default:
+      console.error(`Unsupported geometry type: ${geometryType}`);
+  }
+}
+
+// Primitives; parse point and linear ring
+function parsePoint(view, offset, dimension, littleEndian) {
+  const positions = new Float64Array(dimension);
+  for (let i = 0; i < dimension; i++) {
+    positions[i] = view.getFloat64(offset, littleEndian);
+    offset += 8;
+  }
+
+  return {positions: {value: positions, size: dimension}, offset};
+}
+
+function parseLineString(view, offset, dimension, littleEndian) {
+  const nPoints = view.getUint32(offset, littleEndian);
+  offset += 4;
+
+  // Instantiate array
+  const positions = new Float64Array(nPoints * dimension);
+  for (let i = 0; i < nPoints * dimension; i++) {
+    positions[i] = view.getFloat64(offset, littleEndian);
+    offset += 8;
+  }
+
+  return {
+    positions: {value: positions, size: dimension},
+    pathIndices: {value: new Uint16Array([0, nPoints]), size: 1},
+    offset
+  };
+}
+
+// https://stackoverflow.com/a/55261098
+const cumulativeSum = sum => value => (sum += value);
+
+function parsePolygon(view, offset, dimension, littleEndian) {
+  const nRings = view.getUint32(offset, littleEndian);
+  offset += 4;
+
+  const rings = [];
+  for (let i = 0; i < nRings; i++) {
+    var {positions, offset} = parseLineString(view, offset, dimension, littleEndian);
```

Comment:
Probably best to not destructure, since I want to update `offset`, but declare `positions` as `const`